app/models/pdc_metadata/related_object.rb

```diff
@@ -36,5 +49,23 @@ def self.relation_type_options
       built = Hash[pairs]
       built.with_indifferent_access
     end
+
+    private
+
+      def valid_related_identifier_type?
+        @valid_related_identifier_type ||= valid_type_values.include?(related_identifier_type&.upcase)
+      end
+
+      def valid_relation_type?
+        @valid_relation_type ||= valid_relationship_types.include?(relation_type&.upcase)
+      end
+
+      def valid_type_values
+        @valid_type_values ||= Datacite::Mapping::RelatedIdentifierType.to_a.map(&:value).map(&:upcase)
+      end
+
+      def valid_relationship_types
+        @valid_relationship_types ||= Datacite::Mapping::RelationType.to_a.map(&:value).map(&:upcase)
+      end
```

Comment:
```suggestion
      def valid_related_identifier_type?
        valid_type_values.include?(related_identifier_type&.upcase)
      end

      def valid_relation_type?
        valid_relationship_types.include?(relation_type&.upcase)
      end

      def valid_type_values
        Datacite::Mapping::RelatedIdentifierType.to_a.map(&:value).map(&:upcase)
      end

      def valid_relationship_types
        Datacite::Mapping::RelationType.to_a.map(&:value).map(&:upcase)
      end
```

I don't know the bigger picture, but is the idea just to store the result to avoid recomputation? Unless these are really expensive operations, I'm not sure the additional complexity is worth it.

Reply:
Not needed because of the attr_accessor